wlm/src/main/java/ai/djl/serving/wlm/ModelInfo.java

```diff
@@ -457,22 +458,124 @@ public int getQueueSize() {
         return queueSize;
     }
 
+    /**
+     * Sets the minimum number of workers.
+     *
+     * @param minWorkers the new minimum number of workers
+     */
+    public void setMinWorkers(int minWorkers) {
+        if (maxWorkers != null && maxWorkers < minWorkers) {
+            throw new IllegalArgumentException(
+                    "The max workers for a model can't be smaller than the min workers");
+        }
+
+        this.minWorkers = minWorkers;
+    }
+
     /**
      * Returns the minimum number of workers.
      *
+     * @param device the device to get the min workers for
      * @return the minimum number of workers
      */
-    public int getMinWorkers() {
-        return minWorkers;
+    public int getMinWorkers(Device device) {
+        if (minWorkers != null && minWorkers >= 0) {
```

Comment:
should this be minWorkers > 0?

I think we need at least one worker

Reply:
This is really a check for null and `-1` which are the "use default" values. I can add that validation, although I would add it on the side of setting the minWorkers and maxWorkers. I agree that `maxWorkers > 0`, but there is a case for `minWorkers = 0`. For the use case where you have many CPU models with low usage of each, there are two approaches:

1. Allow `minWorkers == 0` and spin up the worker when the model is first called. 
2. Require the `minWorkers > 0` which would result in large numbers of workers (`workers >> numCpus`) and most of those workers should just remain idle.

Have we compared the two approaches?

Reply:
Ideally, we should allow minWorkers == 0, but it requires we support scale to 0, which we don't currently.